msrest/serialization.py

```diff
@@ -615,9 +615,23 @@ def query(self, name, data, data_type, **kwargs):
         if self.client_side_validation:
             data = self.validate(data, name, required=True, **kwargs)
         try:
-            if data_type in ['[str]']:
-                data = ["" if d is None else d for d in data]
+            # Treat the list aside, since we don't want to encode the div separator
+            if data_type.startswith("["):
+                internal_data_type = data_type[1:-1]
+                data = [
+                    self.serialize_data(d, internal_data_type, **kwargs) if d is not None else ""
+                    for d
+                    in data
+                ]
+                if not kwargs.get('skip_quote', False):
+                    data = [
+                        quote(str(d), safe='')
+                        for d
+                        in data
+                    ]
+                return str(self.serialize_iter(data, internal_data_type, **kwargs))
```

Comment:
I assume we are not removing the capability to have separators other than ',' by calling `serialize_iter`?

Reply:
Ah, never mind, I see we have tests for that. I would probably have called the parameter `sep` or `separator` rather than `div`. I was wondering what HTML had to do with this method when I first saw it :)

Reply:
legacy :)